src/android/com/adobe/phonegap/push/FCMService.java

```diff
@@ -103,8 +103,13 @@ public void onMessageReceived(RemoteMessage message) {
         PushPlugin.setApplicationIconBadgeNumber(getApplicationContext(), 0);
       }
 
+      int clearNotificationId = getClearNotificationId(extras);
+
+      // if the notification has a `clearNotification`, then clear it.
+      if(clearNotificationId != -1) {
+        PushPlugin.clearNotification(getApplicationContext(), clearNotificationId);
       // if we are in the foreground and forceShow is `false` only send data
-      if (!forceShow && PushPlugin.isInForeground()) {
+      } else if (!forceShow && PushPlugin.isInForeground()) {
```

Comment:
The way this is right now, and considering the example you used on the OP, I can still use the `clearNotification` attribute on the payload to clear said id if the application is in foreground, *however*  the scenario on this line won't trigger and it'll fall through to the last `else {`, which will consider foreground:false even though we're in foreground, therefore creating a very inconsistent payload for the js to handle.

So, we have a few options:
- we can consider that the `clearNotification` attribute represents a separate and independent feature on the payload, and a single payload can be {an actual notification with all the other payload attributes available} and {a request to clear a specific notification} *at the same time*. This would need the first `if` statement here to be isolated as to not race against the existing ones. Check the `clearBadge` feature, as that's the closest we have to this option.
- we can consider that the `clearNotification` attribute represents the whole feature, and whenever it is present it will skip all other cases (`if/else` statements). This would need the first `if` statement to actually return from the method, or make the last `else` statement do nothing if it did clear a notification before. This would also make the example used on the OP invalid, as the `title` and `body` attributes on the payload mean nothing. This would also make it useful to move this feature to as early as possible on the method to avoid useless work (if we're gonna clear a notification and return, why waste resources preping the extras? This would also make it needed to solve the question "who has the highest priority on the feature order: `clearBadge` or `clearNotification`?".

Reply:
I would consider going for the first option, I would consider that the most likely use-case for people. And that way you don't force people. Will change and test it :-)